sycl/source/detail/platform_impl.cpp

```diff
@@ -116,6 +114,8 @@ std::vector<platform> platform_impl::get_platforms() {
       for (const auto &PiPlatform : PiPlatforms) {
         platform Platform = detail::createSyclObjFromImpl<platform>(
             getOrMakePlatformImpl(PiPlatform, Plugins[i]));
+        // insert PiPlatform into the Plugin
```

Comment:
```suggestion
        std::lock_guard<std::mutex> Guard(*Plugin.getPluginMutex());
        // insert PiPlatform into the Plugin
```
Shouldn't `getPlatformId` be guarded by `MPluginMutex` according to:

```
  // Mutex to guard PiPlatforms and LastDeviceIds.
  // Note that this is a temporary solution until we implement the global
  // Device/Platform cache later.
  std::shared_ptr<std::mutex> MPluginMutex;
```